### PR TITLE
Add discount models to order

### PR DESCRIPTION
**Pull request by korycins**
I had to open a new PR for these changes. The `push -f` was not my friend during the rebasing the all feature branches related to these changes, which causes me to make mistake.  As a result of this, gh closed my PR and marked it as merged. 

Old PR https://github.com/mirumee/saleor/pull/6874

I want to merge this change because I need to extend the models to better handle the discount logic

FYI: This is the PR only related to the model changes. The workflow for the API should be the same

https://app.clickup.com/t/2549495/SALEOR-1875

# Impact

* [x ] New migrations
* [ ] New/Updated API fields or mutations
* [x ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1875 New model Discount connected with Order](https://app.clickup.com/t/2549495/SALEOR-1875)

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ All notable, unreleased changes to this project will be documented in this file.
 - Drop `CHECKOUT_QUANTITY_CHANGED` webhook - #6797 by @d-wysocki
 - Drop deprecated `taxRate` field from `ProductType` - #6795 by @d-wysocki
 - Remove resolving user's location from GeoIP; drop `PaymentInput.billingAddress` input field - #6784 by @maarcingebala
+- Change the payload of the order webhook to handle discounts list.
+Deprecate fields order.discount, order.discount_name, order.translated_discount_name - #6874 by @korycins
```
Comment (maarcingebala) [inline CHANGELOG.md:21]:
```suggestion
Deprecate fields `Order.discount`, `Order.discountName`, `Order.translatedDiscountName` - #6874 by @korycins
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,8 @@ All notable, unreleased changes to this project will be documented in this file.
 - Drop `CHECKOUT_QUANTITY_CHANGED` webhook - #6797 by @d-wysocki
 - Drop deprecated `taxRate` field from `ProductType` - #6795 by @d-wysocki
 - Remove resolving user's location from GeoIP; drop `PaymentInput.billingAddress` input field - #6784 by @maarcingebala
+- Change the payload of the order webhook to handle discounts list.
```
Comment (maarcingebala) [inline CHANGELOG.md:20]: How will users know what changed exactly?

File: saleor/order/utils.py [hunk 3]
```diff
@@ -65,12 +66,16 @@ def decorator(*args, **kwargs):
...
-            order.discount = discount
-        return func(*args, **kwargs)
+        return func(*args, **kwargs, discount=discount)
 
     return decorator
 
 
+def get_voucher_discount_assigned_to_order(order: Order):
+    voucher_discount = order.discounts.filter(type=OrderDiscountType.VOUCHER).first()
+    return voucher_discount
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/order/utils.py:76]: I am aware that this is single line function, but I would prefer to keep it as function. imo it is more meaningful than and easier to find in future.

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**